ocamltest/environments.ml

```diff
@@ -69,26 +103,29 @@ let safe_lookup variable env = match lookup variable env with
 let is_variable_defined variable env =
   VariableMap.mem variable env
 
-let add variable value env = VariableMap.add variable value env
+let add variable value env = VariableMap.add variable (Some value) env
 
 let add_if_undefined variable value env =
   if VariableMap.mem variable env then env else add variable value env
 
 let append variable appened_value environment =
   let previous_value = safe_lookup variable environment in
   let new_value = previous_value ^ appened_value in
-  VariableMap.add variable new_value environment
+  VariableMap.add variable (Some new_value) environment
 
-let remove = VariableMap.remove
+let remove variable environment =
+  VariableMap.add variable None environment
 
 let add_bindings bindings env =
   let f env (variable, value) = add variable value env in
   List.fold_left f env bindings
 
 let from_bindings bindings = add_bindings bindings empty
 
-let dump_assignment log (variable, value) =
-  Printf.fprintf log "%s = %s\n%!" (Variables.name_of_variable variable) value
+let dump_assignment log = function
+  | (variable, Some value) ->
+    Printf.fprintf log "%s = %s\n%!" (Variables.name_of_variable variable) value
+  | (_, None) -> ()
```

Comment:
`None` means "remove this variable from the environment", right? Shouldn't you print a line in this case? This will make the users' lives easier.

Reply:
Yes - done!

(just for info, this is displayed when you use the `dumpenv` debugging action)